### DISCUSSION TITLE
Best Practices for Cross-Platform Authorization and Role Management in AvaloniaUI

### DISCUSSION THREAD

**Discussion by TraZz-Kami** [AvaloniaUI/Avalonia#13039] [Q&A]
Hello everyone !

I'm working on cross-platform development with AvaloniaUI and need advice on effective authorization and role management. If you have experience or insights to share, please jump in. Your input will help me create a secure and cross-platform application.

Thanks in advance for your help!

**Comment by timunie** [comment]
Potential idea:
- add an interface in the shared project
- implement that interface in Andriod/IOS/Desktop project and handle the Auth there
- Probably Maui.Essentials or Xamarin.Essentials can be used, check these libs :-)

**Comment by thevortexcloud** [comment]
If needs to be all done locally you can probably get something working with SqlLite. It's supported on pretty much everything in some capacity. Although if you are doing completely local authentication I would say it's probably not very effective. If you are using an external provider I imagine the process will be the same for every platform, just the UI might be slightly different. 

If you want to try to set up a server to handle all of this for you, ASP Core Identity is fairly easy to use and is developed by Microsoft. So it's fairly well tested security wise. It's somewhat complicated to use outside web environments though as it's typically used with cookies. But you can easily use it to check credentials and permissions and then use that to issue your own tokens (EG JWT) which can be used for subsequent authentication requests to other endpoints.

Otherwise there are plenty of third party auth providers who will do most of this for you. But you also need to trust that they will keep your user's data secure.

In any case, I would suggest if you can that you avoid collecting things like passwords in the first place.
